rustls/src/client/hs.rs

```diff
@@ -118,27 +113,29 @@ fn find_session(
 pub(super) fn start_handshake(
     dns_name: webpki::DnsName,
     extra_exts: Vec<ClientExtension>,
+    config: &Arc<ClientConfig>,
```

Comment:
Wherever we're not going to clone the `Arc` then should't we use `&ClientConfig` instead of `&Arc<ClientConfig>` to be more general and to clarify the ownership/cloning?

I would expect everything to be either `config: Arc<ClientConfig>` if moving from one state to another, or `config: &ClientConfig` otherwise.

Reply:
`start_handshake()` is actually the entry point to the client handshake now, and it does clone the `Arc` when calling `emit_client_hello_for_retry`.

Reply:
After reviewing the rest, I think this should be `Arc<ClientConfig>`, i.e. cloned by the caller.

Reply:
Not sure I agree. We don't need to clone it sooner, and there is value in not cloning it if we're going to error out of `start_handshake()`, for example.

Reply:
Changed it anyway for consistency.